### DISCUSSION TITLE
New to InfiniTime: Change wrist, reorder apps?

### DISCUSSION THREAD

**Discussion by IngwiePhoenix** [InfiniTimeOrg/InfiniTime#2205] [General]
I just unpacked and updated (with Gadgetbridge) to the current 1.15 version and I really like the simplicity of the interface.

However, I wear my watch on the left arm, because my right eye is blind. So I would like to flip the default orientation.

Further, is there a way to reorder apps? I don't need many, so it would be great to reorder them or make individual icons bigger to make it easier to see.

Thank you and kind regards!

**Comment by vkareh** [comment]
Not sure I follow: watches can be worn on either wrist without having to reorient. I've always worn watches on my left wrist.

With regards to the apps, you'd need to recompile the software to reorder the apps. I do that myself to reorder the ones I use and to remove the ones I don't.

**Reply by IngwiePhoenix** [reply]
Ah, let me rephrase: Basically if I wear it on the left, the button is on the left side of the display - and I'd like it on the right.
I am not a native, sometimes my blabbing doesn't make sense, sorry. ^^;

Alright then, I'll grab the deps in my WSL then. That's fine and fair - its super small so this shouldn't be a biggie. Thanks!

**Reply by vkareh** [reply]
Ah of course! Good point on the button position! I've had to adapt to using the button on the left 😛 
Hopefully this ends up working for you. Not sure how hard it would be to flip the entire display.

For the apps, here are some instructions for enabling/disabling. The apps show up in the order you define them in that variable: https://github.com/InfiniTimeOrg/InfiniTime/blob/d69cfcfb136e45486437ad773d362da7870042e3/doc/code/Apps.md?plain=1#L192

**Reply by IngwiePhoenix** [reply]
I was just done reading the DisplayApp :) I have some experience with the LVGL API but not RTOS - and I am not a huge fan of C++...so, this is taking me a little longer than I'd like lol.

Thank you for the pointer! Much appreciated.

Since this isn't a feature, and I didn't find any real mention elsewhere here, might as well post this as a proper feature request. Technically, on the root surface, you'd just flip the draw order from top-to-bottom to bottom-to-top. But, that depends on the display driver, so I'll have to read that and eeeeh, i'll find it eventually to possibly help contribute lol.

**Reply by mark9064** [reply]
I wear it on the left and find the left button to be comfortable, I press it with my thumb and brace the right side of the watch with my index finger. But I imagine you've tried that already

Definitely take a look at the closed pr #98 , there might be some handy stuff there. Importantly the hardware scrolling will be the most annoying bit to get right. Another possible strategy (compared to LVGL draw order) is changing the display scanout direction in the LCD, the datasheet is pretty comprehensive so it should cover if it's possible. Hardware scrolling may be better this way? - I'm not sure

**Reply by duckunix** [reply]
While I wear my watched on the left, the issue I have with the button on the left is that it seems to get pushed by things like my shirt cuff or bedding while I sleep.  Also, it is the only watch I have owned with the 'crown' on the left vs. right side.
